### PR TITLE
Improve compatibility page to include history and SemVer score

### PR DESCRIPTION
**Pull request by petehamilton**
After this, users can add a badge to their site displaying their SemVer
score, which links to a page explaining what it means.

The compatibility scores that Dependabot already shows in PRs also now
link to an updated screen, following the same design.

---

*New default compatibility page/dependency without version*
image

*New dependency semver page and badge*
image

*New dependency compatibility (specific version bump, linked from dependabot PR)*
image

*New dependency compatibility (not enough data)*
image

*New dependency semver (not enough data)*
image

### CODE REVIEW

File: css/main.css [hunk 1]
```diff
@@ -1,38 +1,24 @@
-/*! normalize.css v7.0.0 | MIT License | github.com/necolas/normalize.css */
+/*! normalize.css v8.0.0 | MIT License | github.com/necolas/normalize.css */
```
Comment (petehamilton) [inline css/main.css]: I believe this is auto-generated?

Comment (greysteil) [reply css/main.css]: Yep - fine to include.

File: compatibility-score.html [hunk 2]
```diff
@@ -174,7 +132,160 @@ <h3>Avoid buggy releases</h3>
...
+    }
+
+    function semverScore(total, successful) {
+      return Math.round(100.0 * successful / total);
+    }
+
+    function compatibilityBadgeUrl(dependencyName, packageManager, previousVersion, newVersion) {
+      // If we're not supplying versions, get a badge for the entire dependency history
+      // Otherwise, get a badge for just the version bump we care about
+      if (!previousVersion && !newVersion) {
```
Comment (greysteil) [inline compatibility-score.html]: This should be `||` - you need to pass both to the backend if either is present.

Comment (petehamilton) [reply compatibility-score.html]: Updated, I flipped the condition and the blocks to make the intent a little clearer:

```
if (prev + new) {
  badge_with_version
} else {
  without
}
```

etc

**Pull request merged**